operator/internal/manifests/node_placement.go

```diff
@@ -78,3 +95,40 @@ func configureReplication(podTemplate *corev1.PodTemplateSpec, replication *loki
 
 	return nil
 }
+
+func initContainerAZAnnotationCheck(image string) corev1.Container {
+	azPath := fmt.Sprintf("%s/%s", availabilityZoneVolumeMountPath, availabilityZoneVolumeFileName)
+	return corev1.Container{
+		Name:  "az-annotation-check",
+		Image: image,
+		Command: []string{
+			"sh",
+			"-c",
+			fmt.Sprintf("while ! [ -s %s ]; do echo Waiting for availability zone annotation to be set; sleep 2; done; echo availability zone annotation is set; cat %s", azPath, azPath),
```

Comment:
As far as I can see, the generated file for the annotation will (correctly) not contain a newline at the end.

This might look weird in output and mess with logging systems if the last thing the init container emits is a line with no EOL. I suggest we add another newline after outputting the AZ.